src/components/export-select-fields/export-select-fields.jsx

```diff
@@ -0,0 +1,73 @@
+import { Tooltip } from 'hadron-react-components';
+import ExportField from 'components/export-field';
+import React, { Component } from 'react';
+import isEqual from 'lodash.isequal';
+import PropTypes from 'prop-types';
+
+
+import styles from './export-select-fields.less';
+import createStyler from 'utils/styler.js';
+const style = createStyler(styles, 'export-select-fields');
+
+const fieldInfoSprinkle = 'The fields displayed are from a sample of documents in the collection. To ensure all fields are exported, add missing field names.';
+
+class ExportSelectFields extends Component {
+  static propTypes = {
+    fields: PropTypes.object.isRequired,
+    updateFields: PropTypes.func.isRequired,
+  };
+
+  shouldComponentUpdate(nextProps) {
```

Comment:
@imlucas would you by any chance know how to get a component to rerender when there is a change in a nested prop? In this case one of the items in `fields` was changed and I want to make sure this rerenders. I can't get this function to run at all o.O